dg.go

```diff
@@ -24,20 +25,32 @@ type directedGraph[NodeType any] struct {
 }
 
 func (d *directedGraph[NodeType]) Mermaid() string {
-    result := []string{}
-    result = append(result, "flowchart TD")
-    result = append(result, "subgraph input")
-    for source := range d.connectionsFromNode {
-	if strings.HasPrefix(source, "input") {
-            result = append(result, source)
-	}
+    result := []string{
+        "%% Mermaid markdown workflow",
+        "flowchart LR",
+        "%% Success path",
     }
-    result = append(result, "end")
+
+    errorPath := []string{"%% Error path"}
+    errorPathRegex, _ := regexp.Compile("error|crashed|failed|deploy_failed")
+
     for source, d := range d.connectionsFromNode {
         for destination := range d {
-            result = append(result, fmt.Sprintf("%s-->%s", source, destination))
+            destinationNodes := strings.Split(destination, ".")
+            match := errorPathRegex.MatchString(
```

Comment:
I don't think this variable name is a good one. I'd do something like `isErrorPath`

Reply:
Done.